Source/santad/Santad.mm

```diff
@@ -0,0 +1,281 @@
+}
+
+void SantadMain(std::shared_ptr<EndpointSecurityAPI> esapi, std::shared_ptr<Logger> logger,
+                std::shared_ptr<Metrics> metrics, std::shared_ptr<Enricher> enricher,
+                std::shared_ptr<AuthResultCache> auth_result_cache,
+                MOLXPCConnection *control_connection, SNTCompilerController *compiler_controller,
+                SNTNotificationQueue *notifier_queue, SNTSyncdQueue *syncd_queue,
+                SNTExecutionController *exec_controller,
+                std::shared_ptr<SNTPrefixTree> prefix_tree) {
+  SNTConfigurator *configurator = [SNTConfigurator configurator];
+
+  SNTDaemonControlController *dc =
+    [[SNTDaemonControlController alloc] initWithAuthResultCache:auth_result_cache
+                                              notificationQueue:notifier_queue
+                                                     syncdQueue:syncd_queue
+                                                         logger:logger];
+
+  control_connection.exportedObject = dc;
+  [control_connection resume];
+
+  if ([configurator exportMetrics]) {
+    metrics->StartPoll();
+  }
+
+  SNTEndpointSecurityDeviceManager *device_client =
+    [[SNTEndpointSecurityDeviceManager alloc] initWithESAPI:esapi
+                                                     logger:logger
+                                            authResultCache:auth_result_cache];
+
+  device_client.blockUSBMount = [configurator blockUSBMount];
+  device_client.remountArgs = [configurator remountUSBMode];
+  device_client.deviceBlockCallback = ^(SNTDeviceEvent *event) {
+    [[notifier_queue.notifierConnection remoteObjectProxy]
+      postUSBBlockNotification:event
+             withCustomMessage:([configurator remountUSBMode]
+                                  ? [configurator bannedUSBBlockMessage]
+                                  : [configurator remountUSBBlockMessage])];
+  };
+
+  SNTEndpointSecurityRecorder *monitor_client =
+    [[SNTEndpointSecurityRecorder alloc] initWithESAPI:esapi
+                                                logger:logger
+                                              enricher:enricher
+                                    compilerController:compiler_controller
+                                       authResultCache:auth_result_cache
+                                            prefixTree:prefix_tree];
+
+  SNTEndpointSecurityAuthorizer *authorizer_client =
+    [[SNTEndpointSecurityAuthorizer alloc] initWithESAPI:esapi
+                                          execController:exec_controller
+                                      compilerController:compiler_controller
+                                         authResultCache:auth_result_cache];
+
+  SNTEndpointSecurityTamperResistance *tamper_client =
+    [[SNTEndpointSecurityTamperResistance alloc] initWithESAPI:esapi logger:logger];
+
+  EstablishSyncServiceConnection(syncd_queue);
+
+  NSArray<SNTKVOManager *> *kvoObservers = [[NSArray alloc]
+    initWithObjects:
```

Comment:
I think this would be slightly more readable initializing all the managers into variables and making the array of them at the end. Also using an array literal would be neater.

Reply:
Can change to variables if preferred. I initially did that, but didn't like a bunch of unnecessary variables hanging around. The array here was used so only a single "unused" variable was needed instead of the handful of individual ones.